PRIS-177: docs: add RELEASING.md and rewrite README (merge after #168) (#170)
OpenNMS/opennms-provisioning-integration-server: README.md

# OpenNMS Provisioning Integration Server (PRIS)

[![CI](https://github.com/OpenNMS/opennms-provisioning-integration-server/actions/workflows/ci.yml/badge.svg)](https://github.com/OpenNMS/opennms-provisioning-integration-server/actions/workflows/ci.yml)
[![Latest release](https://img.shields.io/github/v/release/OpenNMS/opennms-provisioning-integration-server?sort=semver)](https://github.com/OpenNMS/opennms-provisioning-integration-server/releases)

> [![Container image](https://img.shields.io/badge/ghcr.io-opennms%2Fpris-2496ED?logo=docker&logoColor=white)](https://github.com/OpenNMS/opennms-provisioning-integration-server/pkgs/container/pris)

Reviewer: ```suggestion
[![Container image](https://img.shields.io/badge/docker.io-opennms%2Fpris-2496ED?logo=docker&logoColor=white)](https://hub.docker.com/r/opennms/pris)
```

[![Java 17](https://img.shields.io/badge/Java-17-orange.svg?logo=openjdk&logoColor=white)](https://adoptium.net/)
[![License: GPL v3](https://img.shields.io/badge/License-GPLv3-blue.svg)](LICENSE)

PRIS gets external information from your inventory into an OpenNMS requisition
model. It serves the result as XML over HTTP so OpenNMS Provisiond can import
and discover nodes from it, and lets you normalize, clean up and manipulate the
inventory data before it reaches OpenNMS.

## Contents

- [Quick start (container)](#quick-start-container)
- [Documentation](#documentation)
- [Build from source](#build-from-source)
- [Build a container image locally](#build-a-container-image-locally)
- [Project layout](#project-layout)
- [Contributing and releasing](#contributing-and-releasing)
- [Project information](#project-information)
- [License](#license)

## Quick start (container)

> Released images are published to the [GitHub Container Registry]. Start the

Reviewer: ```suggestion
Released images are published to [Docker Hub]. Start the
```

latest stable release with:

```sh

> docker run --name mypris --detach --publish 8000:8000 ghcr.io/opennms/pris:latest

Reviewer: ```suggestion
docker run --name mypris --detach --publish 8000:8000 opennms/pris:latest
```

```

Then open <http://localhost:8000>. If no requisition source is configured yet
you are redirected to the documentation page.

Image tags:

- `latest` — floating tag tracking the most recent stable release

> - `<version>` — a specific release (see the [container package][GitHub Container Registry] for available tags)

Reviewer: ```suggestion
- `<version>` — a specific release (see the [Docker Hub repository][Docker Hub] for available tags)
```

### Docker Compose

```yaml
volumes:
  pris.data:
    driver: local

services:
  pris:
    container_name: opennms.pris

>     image: ghcr.io/opennms/pris:latest

Reviewer: ```suggestion
    image: opennms/pris:latest
```

`make oci IMAGE=mypris VERSION=1.2.3`, then run it:

```sh
docker run --name mypris --detach --publish 8000:8000 "pris:$(cat version.txt)"
```

Multi-arch (amd64/arm64) images are built and published only by the release
workflow — see [RELEASING.md](RELEASING.md).

## Project layout

Maven multi-module project:

| Module | Purpose |
| --- | --- |
| `opennms-pris-api` | Programming interfaces for configuration, mappers and data sources |
| `opennms-pris-model` | The OpenNMS requisition model |
| `opennms-pris-main` | The provisioning integration server itself |
| `opennms-pris-plugins` | Data-source plugins: XLS, script, JDBC, OCS Inventory and defaults |
| `opennms-pris-dist` | Assembles the compiled code into a runnable, distributable archive |

User and developer documentation lives under `docs/` and is built with [Antora].

## Contributing and releasing

- CI/CD workflows live in [`.github/workflows`](.github/workflows); every push
  and pull request is built, packaged and smoke tested.
- The release process (tag-driven, publishing container images and archives) is
  documented in **[RELEASING.md](RELEASING.md)**.

## Project information

- CI/CD system: [GitHub Actions]

> - Container images: [GitHub Container Registry]

Reviewer: ```suggestion
- Container images: [Docker Hub]
- ```

- Issue and bug tracking: [GitHub Issues]
- Source code: [GitHub]
- Chat: [Web Chat]
- Maintainer: ronny@opennms.org

## License

Distributed under the GNU General Public License v3.0. See [LICENSE](LICENSE) for
the full text.

[GitHub]: https://github.com/OpenNMS/opennms-provisioning-integration-server
[GitHub Actions]: https://github.com/OpenNMS/opennms-provisioning-integration-server/actions

> [GitHub Container Registry]: https://github.com/OpenNMS/opennms-provisioning-integration-server/pkgs/container/pris

Reviewer: ```suggestion
[Docker Hub]: https://hub.docker.com/r/opennms/pris
```
